dot/peerset/peerset.go

```diff
@@ -586,26 +622,42 @@ func (ps *PeerSet) incoming(setID int, peers ...peer.ID) error {
 		}
 
 		state := ps.peerState
-		p := state.nodes[pid]
+
+		var nodeReputation Reputation
+
+		state.RLock()
+		node, has := state.nodes[pid]
+		if has {
+			nodeReputation = node.reputation
+		}
+		state.RUnlock()
+
 		switch {
-		case p.getReputation() < BannedThresholdValue:
-			ps.resultMsgCh <- Message{
-				Status: Reject,
-				setID:  uint64(setID),
-				PeerID: pid,
-			}
-		case state.tryAcceptIncoming(setID, pid) != nil:
+		case nodeReputation < BannedThresholdValue:
 			ps.resultMsgCh <- Message{
 				Status: Reject,
 				setID:  uint64(setID),
 				PeerID: pid,
 			}
+
 		default:
-			logger.Debugf("incoming connection accepted from peer %s", pid)
-			ps.resultMsgCh <- Message{
-				Status: Accept,
-				setID:  uint64(setID),
-				PeerID: pid,
+			err := state.tryAcceptIncoming(setID, pid)
+			if err != nil {
+				logger.Errorf("cannot accept incomming peer %pid: %w", pid, err)
+
+				ps.resultMsgCh <- Message{
+					Status: Reject,
+					setID:  uint64(setID),
+					PeerID: pid,
+				}
+			} else {
+				logger.Debugf("incoming connection accepted from peer %s", pid)
+
+				ps.resultMsgCh <- Message{
+					Status: Accept,
+					setID:  uint64(setID),
+					PeerID: pid,
+				}
```

Comment:
nit can you instead set the message in each test case:

```go
message = Message{...}
```

and then send it in the channel at the end of the `switch` with `ps.resultMsgCh <- message`

You can even pre-set a common message with the peer id and set id to reduce the amount of lines/repetition.